### ISSUE TITLE
Documentation and advancement

### ISSUE THREAD

**Issue by crichon** [justinclayton/packer-builder-vsphere#1]
Hi, I stumble upon your proposal: [582](https://github.com/mitchellh/packer/pull/582) on mitchellh/packer which led me here.

I have tried to build this repository but failed. I ended up with:

```
vsphere/builder.go:93: cannot use artifact (type *Artifact) as type packer.Artifact in return argument
```

Excuse me if I have missed something obvious but I am not use to go yet. 

I need such a builder so I would like to know wherever you were still working on it. In case you are not, would you mind helping me taking over and finish it.

Thanks.

**Comment by justinclayton** [comment]
Absolutely! It was working with the latest version of packer at the time, but that was months ago, so I'm not surprised that it needs a few updates. It also should have tests written if it's ever to be merged into core packer. I'd love to work with you on this.

DMing me on Twitter is probably the best way to contact me outside of github.

**Comment by crichon** [comment]
Thx for this quick reply ;) I am all yours.

Indeed it would be nice beeing merged into core packer. I will take some time to go through your codebase. By any chance, would you have time to update it ? It would be greatly appreciated.
